pkg/proxy/proxy.go

```diff
@@ -97,6 +97,9 @@ type ProxySource interface {
 
 // Proxy maintains state about redirects
 type Proxy struct {
+	// proxyKind specifies which proxy is being used.
+	proxyKind string
```

Comment:
Add a note that this only has effect for protocols that have multiple proxy implementations.